src/Reflection/Adapter/ReflectionUnionType.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection\Adapter;
+
+use ReflectionUnionType as CoreReflectionUnionType;
+use Roave\BetterReflection\Reflection\ReflectionType as BetterReflectionType;
+use Roave\BetterReflection\Reflection\ReflectionUnionType as BetterReflectionUnionType;
+
+use function array_map;
+
+class ReflectionUnionType extends CoreReflectionUnionType
+{
+    private BetterReflectionUnionType $betterReflectionType;
+
+    public function __construct(BetterReflectionUnionType $betterReflectionType)
+    {
+        $this->betterReflectionType = $betterReflectionType;
+    }
+
+    /**
+     * @return array<ReflectionNamedType|ReflectionType|ReflectionUnionType|null>
```

Comment:
I think this should return a `list`, not an `array` 🤔

Reply:
I would also expect a `non-empty-list`, which will force further refining of the BR implementation too